src/MongoBulkDataMigration.ts

```diff
@@ -158,24 +158,22 @@ export default class MongoBulkDataMigration<TSchema extends Document>
       updatePromises.push(bulkUpdateWrappedPromise);
 
       document = (await cursor.next()) as WithId<TSchema> | null;
-      const totalIncludingPendingBulks =
-        bulkBackup.size +
-        updatePromiseLimiter.activeCount +
-        updatePromiseLimiter.pendingCount;
-      if (!document || totalIncludingPendingBulks >= this.options.maxBulkSize) {
```

Comment:
> The previous computation used `bulkBackup.size`, which is always 0

@LucVidal360 This is not true when `rollbackable:true` (= most of the time)

I must admit at this moment I don't understand this code: why did I introduced count of `updatePromiseLimiter` promises limiter :thinking: 
Maybe there should be a logic with `rollbackable` here instead?

Reply:
@pp0rtal Of course. I meant, `bulkBackup.size` is always 0 *when rollbackable is false*.

I'm using directly `updatePromises.lenght` here, as it's a better way to reflect the amount of updates we've already taken (with possibly some promises not resolved yet).

Before, the code counted:
- the amount of promises already resolved via the number of rollback documents created
- + the amount of unresolved promises (active + pending)

To reassure yourself, I found some courage, and added tests with and without rollbackable: 1c45cd2